tools/BVH/BVHEngine.cpp

```diff
@@ -45,15 +45,22 @@ BVHEngine::Ptr BVHEngine::create(const std::string& engine_name, size_t dim) {
 #endif
 #if WITH_IGL
     if (engine_name == "igl") {
-        switch (dim) {
-            case 2:
-                return std::make_shared<IGL::AABBTree<2>>();
-            case 3:
-                return std::make_shared<IGL::AABBTree<3>>();
-            default:
-                throw NotImplementedError("Only 2D and 3D meshes are supported");
+        if (dim != 3) {
+            throw NotImplementedError(
+                    "Only 3D meshes are supported by IGL BVH");
```

Comment:
Thanks for the contribution!  2D mesh is not supported because libigl's signed distance function is 3D only?

Reply:
Thanks for the response! When taking AABBtree as an argument, Libigl's signed distance function requires different number of arguments for 2D and 3D cases. I ended up with a compiling error if I considered both 2D and 3D, so I only kept 3D. We might be able to enable both 2D and 3D if 2D is really important for us. Then, we have to re-design some of the PyMesh interfaces, e.g. BVH's "look_up".

Reply:
Got it!  Thanks!

Reply:
Thanks for accepting my pull requests.